File: src/user-input/clean-order.ts
// Third-Party libraries
import inquirer from 'inquirer'

// NodeJS
import process from 'node:process'

// Utils
import { writeMessage } from '@/utils/console'
import { getErrorMessage } from '@/utils/errors'

export async function addStylelintConfigCleanOrder(): Promise<boolean> {
  try {
    const { addStylelintConfigCleanOrder } = await inquirer.prompt({
      type: 'confirm',
      default: true,
      message: 'Do you wanna add stylelint-config-clean-order?:',
      name: 'addStylelintConfigCleanOrder'
    })

    return addStylelintConfigCleanOrder
  } catch {
    writeMessage({
      type: 'error',
      message: getErrorMessage('StylelintConfigCleanOrder')
    })

    process.exit(1)
  }
}

File: user_input/clean-order.go
package userinput

import (
	"os"
	"stylelintbc/utils"

	"github.com/AlecAivazis/survey/v2"
)

func AddStylelintConfigCleanOrder() bool {
	var questions = []*survey.Question{
		{
			Name: "addCleanOrder",
			Prompt: &survey.Confirm{
				Message: "Do you wanna add stylelint-config-clean-order?",
				Default: true,
			},
		},
	}

	answers := struct {
		AddConfigCleanOrder bool `survey:"addCleanOrder"`
	}{}

	err := survey.Ask(questions, &answers)

	if err != nil {
		utils.WriteMessage(utils.WriteMessageProps{
			Type:    "error",
			Message: string(utils.GetErrorMessage("StylelintConfigCleanOrder")),
		})

		os.Exit(1)
	}

	return answers.AddConfigCleanOrder
}
